griptape/tools/google_gmail_client/tool.py

```diff
@@ -44,10 +60,27 @@ def create_draft_email(self, params: dict) -> InfoArtifact | ErrorArtifact:
             )
 
             message = EmailMessage()
-            message.set_content(values["body"])
-            message['To'] = values["to"]
+            message.set_content(body)
+            message['To'] = to_address
             message['From'] = self.owner_email
-            message['Subject'] = values["subject"]
+            message['Subject'] = subject
+
+            # Fetch attachment data from memory
+            if memory_name and artifact_namespace:
+                memory = self.find_input_memory(memory_name)
+
+                if memory:
+                    list_artifacts = memory.load_artifacts(artifact_namespace)
+                    if list_artifacts:
+                        for artifact, attachment_name in zip(list_artifacts.value, attachment_names):
+                            file_data = artifact.value.encode()
+                            message.add_attachment(file_data, maintype='application',
+                                                   subtype='octet-stream', filename=attachment_name)
+                    else:
+                        logging.error(f"Artifact with namespace {artifact_namespace} not found.")
+                        return ErrorArtifact(f"Artifact with namespace {artifact_namespace} not found.")
+                else:
+                    return ErrorArtifact(f"memory not found.")
```

Comment:
Similar logic/nesting here. What do you think about generalizing the logic and moving to a util function? Maybe in `BaseTool`?

Reply:
I don't think it needs to be in the BaseTool since the logic doesn't get used by any other tool. But protected methods seems better to me.